src/wrapper_array.jl

```diff
@@ -176,3 +176,43 @@ end
 @propagate_inbounds function Base.setindex!(a::NamedDimsArray, value, inds...)
     return setindex!(parent(a), value, inds...)
 end
+
+############################################
+# show
+
+# 2-arg show, mainly for repr():
+function Base.show(io::IO, A::NamedDimsArray{L,T,N}) where {L,T,N}
+    print(io, "NamedDimsArray(")
+    show(io, parent(A))
+    print(io, ", ", N == 1 ? QuoteNode(L[1]) : L, ")")
+end
+
+# called by summary(), for main REPL printing:
+function Base.showarg(io::IO, A::NamedDimsArray{L,T,N}, outer) where {L,T,N}
+    print(io, "NamedDimsArray(")
+    Base.showarg(io, parent(A), false)
+    print(io, ", ", N == 1 ? QuoteNode(L[1]) : L, ")")
+end
+
+function Base.print_matrix(io::IO, A::NamedDimsArray)
+    s1 = string("↓ :", dimnames(A,1)) * "  "
+    if ndims(A)==2
+        s2 = string(" "^length(s1), "→ :", dimnames(A,2), "\n")
+        print(io, s2)
+    end
+    ioc = IOContext(io, :displaysize => displaysize(io) .- (1, 0))
+    Base.print_matrix(ioc, parent(A), s1)
+end
+
+if VERSION > v"1.6.0-DEV.1561" # 809f27c53df7a54388a687a847e9494e0d29bd4f
+
+    function Base._show_nd_label(io::IO, A::NamedDimsArray, idxs)
+        print(io, "[:, :, ")
+        for i in 1:length(idxs)
+            sy = dimnames(A, i+2)
+            sy == :_ ? print(io, idxs[i]) : print(io, sy, " = ", idxs[i])
```

Comment:
ah, an unfortunate thing here is that this was show "invalid" indexing. 
e.g. 
```julia
julia> nda = NamedDimsArray(rand(1, 1, 2, 2), (:a, :b, :c, :d))
1×1×2×2 NamedDimsArray(::Array{Float64, 4}, (:a, :b, :c, :d)):
[:, :, c = 1, d = 1] =
      → :b
↓ :a  0.4570253497361896

[:, :, c = 2, d = 1] =
      → :b
↓ :a  0.017185058597076752

[:, :, c = 1, d = 2] =
      → :b
↓ :a  0.9711593270392489

[:, :, c = 2, d = 2] =
      → :b
↓ :a  0.7351655482073511
```
which shows 
```julia
[:, :, c = 2, d = 2] =
      → :b
↓ :a  0.7351655482073511
```
which is not valid indexing syntax:
```julia
julia> nda[:, :, c = 2, d = 2]
ERROR: MethodError: no method matching getindex(::NamedDimsArray{(:a, :b, :c, :d), Float64, 4, Array{Float64, 4}}, ::Colon, ::Colon; c=2, d=2)
Closest candidates are:
  getindex(::NamedDimsArray, ::Any...) at /Users/npr/repos/NamedDims.jl/src/wrapper_array.jl:151 got unsupported keyword arguments "c", "d"
```

I see no reason not to show valid code whenever we can
i.e. in this case to show
```julia
[a = :, b = :, c = 2, d = 2] =
      → :b
↓ :a  0.7351655482073511
```

I also think removing the whiespace around the keyword syntax would help with readability here,
e.g.
```julia
[a=:, b=:, c=2, d=2] =
      → :b
↓ :a  0.7351655482073511
```

I can even see the case for dropping the leading colon-selected dimensions `[:, :`, and just showing
e.g.
```julia
[c=2, d=2] =
      → :b
↓ :a  0.7351655482073511
```

Reply:
Once you have `NamedDimsArray(rand(3,3,1,1),(:x,:_,:_,:_))` then `ans[_=1, _=1]` is still not going to be working code. I didn't think to try omitting the leading `[:, :, `.

Agree that omitting spaces looks better for one-letter names. For longer names  I'm less sure. But I do think repeatinng the names starts to look clumsy:
```
[:, :, channel = 1, year = 1] =
              → :observer
↓ :iteration  0.226816  0.268232   0.507167
              0.642586  0.898197   0.467779
              0.744658  0.0212484  0.098098

# vs.

3×3×1×1 NamedDimsArray(::Array{Float64, 4}, (:iteration, :observer, :channel, :year)):
[iteration=:, observer=:, channel=1, year=1] =
              → :observer
↓ :iteration  0.226816  0.268232   0.507167
              0.642586  0.898197   0.467779
              0.744658  0.0212484  0.098098
```

Reply:
> Once you have NamedDimsArray(rand(3,3,1,1),(:x,:_,:_,:_)) then ans[_=1, _=1] is still not going to be working code.

yeah, that's true. But i don't see it as a reason not to show valid code whenever possible.

Thanks for the examples, i think the combination of omitting spaces and omitting the leading dimensions looks nicest to me
```
3×3×1×1 NamedDimsArray(::Array{Float64, 4}, (:iteration, :observer, :channel, :year)):
[channel=1, year=1] =
              → :observer
↓ :iteration  0.226816  0.268232   0.507167
              0.642586  0.898197   0.467779
              0.744658  0.0212484  0.098098
```

Reply:
See what you think of:
```
         108   -16    48   -44  -111   -51    67      -84  -115   -79    11   -12    74  -118
         115  -111     1    73    99    43    24       74   -22  -126     6   -91  -118     0

□[_=1, four=2]
        → :col
↓ :row   -70    -7   -75   116     9    35   106  …    22   -20    93   -49     2  -111  -122
         -70   -93    70   -37   104  -126     2       88   -66    96   -48    70    41    36
         -75   -43  -126   -51    74   105   -28      -65   -38    18    29   114   -24  -115
```

AxisKeys uses `↓→□` for row/col/panel, and it looked a bit sparse without the `[:, :, ` which Base uses.

Reply:
I find the little square quite confusing (and never understand what "panel" means) 

I think maybe just keeping all the names, including the leading dimensions, is maybe best.

And thanks for working on this, I think it'll be a much loved quality of life improvement

Reply:
Not so keen to repeat, but I don't want the 3rd to be mis-read as being the first dimension, where Base puts a `:`... perhaps you should read `□` as a placeholder for the array being indexed, as a reminder that this isn't Base's printing.

Can think a bit, maybe there are other ways.

Reply:
Let's see what @oxinabox thinks :)

Reply:
Two other ideas are
```
julia> view(ndt, :, :, 1:1, 1:2)
30×40×1×2 NamedDimsArray(view(::Array{Int8, 4}, :, :, 1:1, 1:2), (:row, :col, :_, :four)):
𝒩[_=1, four=1]
        → :col
↓ :row    14   -71    19    99   -76    75    55  …   -94    38   -85    -3   118  -108    68
          95  -113   -76   -60  -121   -37    40      -26    29   -29   -22   -91    -5  -105
```
or
```
julia> view(ndt, :, :, 1, 1:2)
30×40×2 NamedDimsArray(view(::Array{Int8, 4}, :, :, 1, 1:2), (:row, :col, :four)):
(…)[four=1]
        → :col
↓ :row    14   -71    19    99   -76    75    55  …   -94    38   -85    -3   118  -108    68
          95  -113   -76   -60  -121   -37    40      -26    29   -29   -22   -91    -5  -105
          -8    22   -53    61   -56    19    -2      -50   -80   107    -5  -118     4   -56
```

Reply:
> which shows
> 
> [:, :, c = 2, d = 2] =
>       → :b
> ↓ :a  0.7351655482073511
> 
> which is not valid indexing syntax:


Should we make that valid indexing? (if so lets open an issue about it)

I kind of think it is acceptable for show, even if we don't
All that have been shown are pretty acceptable to nme.

We can always mess with it later.
Let get this improvement in, and worry about perfection

Reply:
I think mixed named & positional indexing sounds super-confusing!

Have changed it back to start with `[:, :, `, this is the smallest deviation from what Base does, perhaps a good start as you say.